### PR TITLE
Add documentation on subqueries

### PR DESCRIPTION
**Pull request by angrykoala**
This PR adds documentation on how to construct [subqueries](https://neo4j.com/docs/cypher-manual/current/subqueries/)

### CODE REVIEW

File: docs/modules/ROOT/pages/subqueries/call.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+= Call
+
+Cypher link:https://neo4j.com/docs/cypher-manual/current/subqueries/call-subquery/[CALL subqueries] can be created with `new Cypher.Call()`, to do this, a valid query needs to be passed to `Call`. For example:
+
+[source, javascript]
+----
+const dog = new Cypher.Node({ labels: ["Dog"] });
+const person = new Cypher.Node({ labels: ["Person"] });
+
+const dogName=new Cypher.NamedVariable("dogName")
```
Comment (mjfwebb) [inline docs/modules/ROOT/pages/subqueries/call.adoc:12]:
```suggestion
const dogName = new Cypher.NamedVariable("dogName")
```

File: docs/modules/ROOT/pages/subqueries/call.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    WITH this0
+    MATCH (this0:Person)-[this1:HAS_DOG]->(this2:Dog)
+    RETURN this2.name AS dogName
+}
+RETURN dogName
+----
+
+[NOTE]
+====
+Note how the previous example uses `.concat` to concatenate the first `MATCH` statement and the `CALL` clause.
```
Comment (mjfwebb) [inline docs/modules/ROOT/pages/subqueries/call.adoc:64]: I don't see `.concat` used.

**Review by mjfwebb — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/subqueries/call.adoc (hunk 2 above)
Comment (angrykoala) [reply docs/modules/ROOT/pages/subqueries/call.adoc:64]: Good catch, I skipped a line in the example

**Review by lidiazuin — APPROVED** [review]
Some suggestions only :)

**Pull request merged**